### ISSUE TITLE
'glossgroup.element' and 'glossentry.element' are not contributed to the idElements pattern in their modules

### ISSUE THREAD

**Issue by chrispitude** [dita-ot/dita-ot#3181]
## Expected Behavior
In glossgroupMod.rng and glossentryMod.rng, the glossgroup.element and glossentry.element patterns are not contributed to the global idElements pattern.

However, these two elements do have ID attributes:

```
      <define name="glossgroup.attributes">
        <attribute name="id">
          <data type="ID"/>
        </attribute>

      <define name="glossentry.attributes">
        <attribute name="id">
          <data type="ID"/>
        </attribute>

```

## Actual Behavior
They are missing. However, as evidenced by the presence of their element names in the glossgroup.rng topic shell, they should be included:

```
  <div>
    <a:documentation>ID-DEFINING-ELEMENT OVERRIDES</a:documentation>
    <define name="any">
      <zeroOrMore>
        <choice>
          <ref name="idElements"/>
          <element>
            <anyName>
              <except>
                <name>topic</name>
                <name>concept</name>
                <name>glossgroup</name>
                <name>glossentry</name>
                <nsName ns="http://www.w3.org/1998/Math/MathML"/>
              </except>
            </anyName>

```

## Possible Solution
Update the two modules to add those elements to the idElements group.

## Steps to Reproduce
Search the following OASIS files for 'idElements' and note that it is not defined:

```
./rng/technicalContent/rng/glossgroupMod.rng
./rng/technicalContent/rng/glossentryMod.rng

```
## Environment
DITA-OT version 3.1.3, running in Windows 10 WSL (Ubuntu)

**Comment by robander** [comment]
@drmacro does this look like an issue with the DITA 1.3 grammar files?

**Comment by drmacro** [comment]
The issue report is accurate. However, the use of the idElements pattern is not consistent generally, so this issue is not limited to glossentry and glossgroup--a quick check reveals that referenceMod.rng doesn't contribute to it either.

It's been long enough that I don't remember the details underlying this implementation approach but it is clearly not essential to the correct functioning of the grammars since all the TC-provided shells validate. 

I would log this as something to clean up in DITA 2.0.

**Comment by chrispitude** [comment]
If there's no barrier to doing it earlier, I'd be appreciative. I'm
developing a utility to automate the creation of grammars (specializations,
shells, etc.) based on a human-readable configuration file. My utility
relies on idElements to construct the namespace exceptions in its output,
but the output fails RelaxNG validation when these problematic elements are
included.

I can hack in exceptions to my code, but gosh, you guys have gone through
all the trouble to make this idElements stuff useful for just such a
purpose... :)

- Chris

**Comment by drmacro** [comment]
Remember that the RNGs are a product of the DITA TC, not the DITA Open Toolkit.

Changing the RNGs would require an errata to the DITA 1.3 spec and unless there's actual hard-stop bug in the grammars the TC is not likely to do that.

I understand the frustration with an aspect of automation not being as easy as it could be but as special case processing goes it shouldn't be that big of a deal--the number of elements with ID-type  attributes is quite small and will always include specializations of topic.

You could also apply logic similar to that used by the @domains attribute constructing XQuery to construct the ID stuff in a shell, now that I think about it.

**Comment by robander** [comment]
The update in #3617 includes the fix for this in the draft DITA 2.0 grammar files. As Eliot mentioned, there is not going to be an update to the DITA 1.3 grammar files, so we expect those to remain as they are now.

**Comment by robander** [comment]
Fixed for DITA 2.0 grammars with #3617
